Source/CarthageKit/Xcode.swift

```diff
@@ -172,7 +172,12 @@ public struct BuildArguments {
 		// Disable code signing requirement for all builds
 		// Frameworks get signed in the copy-frameworks action
 		args += [ "CODE_SIGNING_REQUIRED=NO", "CODE_SIGN_IDENTITY=" ]
-
+		
+		// If we're in xcode phase, sanitize
+		let environment = NSProcessInfo.processInfo().environment
+		if let xcodeVersionActual = environment["XCODE_VERSION_ACTUAL"] as? String {
```

Comment:
I from time to times use external arguments. Also, if we compare to other build systems/dependency manager (for ex npm), they all inherit from the current environment. I have mixed feelings about that for that reason.